layers/meta-resin-genericx86/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner

```diff
@@ -25,4 +25,10 @@ if [ "$DURING_UPDATE" = "1" ]; then
 			fi
 		fi
 	fi
+
+	# make sure the bootstrap code (boot.img) is removed in case we are using EFI boot
+	if [ -d /sys/firmware/efi ] ; then
+		device="/dev/"$(findmnt --noheadings --canonicalize --output SOURCE /mnt/boot/ | xargs lsblk -no pkname)
+		dd if=/dev/zero of=$device bs=446 count=1
```

Comment:
Could we ensure that if `$device` is not found correctly, then no dd is run? E.g. wrapping it up like:

```
if [ "$device" != "/dev/" ]; then
   dd .....
fi
```